src/path_converters.h

```diff
@@ -339,74 +347,69 @@ class PathClipper
         if (m_do_clipping) {
             /* This is the slow path where we actually do clipping */
 
-            if (m_end_poly) {
-                m_end_poly = false;
-                return (agg::path_cmd_end_poly | agg::path_flags_close);
-            }
-
-            if (m_has_next) {
-                m_has_next = false;
-                *x = m_nextX;
-                *y = m_nextY;
-                return agg::path_cmd_line_to;
+            if (queue_pop(&code, x, y)) {
+                return code;
             }
 
             while ((code = m_source->vertex(x, y)) != agg::path_cmd_stop) {
-                if (code == (agg::path_cmd_end_poly | agg::path_flags_close)) {
+                switch (code) {
+                case (agg::path_cmd_end_poly | agg::path_flags_close):
```

Comment:
what does this line do? My attempts to understand this by writing code have not gone well

```c++

// switch_statement1.cpp
#include <iostream>

enum test_e
  {
    A, B, C
  };



void helper(test_e inp)
{
  switch (inp)
  {
  case (A | B):
    std::cout<< "One"<< std::endl;
    break;
  case C:
    std::cout<< "Three"<< std::endl;
    break;

  default:
    std::cout<< "NONE?!"<< std::endl;
  }
}

int main() {
  helper(A);
  helper(B);
  helper(C);

}

```

gives

```
22:07 $ ./a.out
NONE?!
One
Three
```